src/nvim/option.c

```diff
@@ -4409,110 +4398,133 @@ int makeset(FILE *fd, int opt_flags, int local_only)
 /// 'sessionoptions' or 'viewoptions' contains "folds" but not "options".
 int makefoldset(FILE *fd)
 {
-  if (put_setstring(fd, "setlocal", "fdm", &curwin->w_p_fdm, 0) == FAIL
-      || put_setstring(fd, "setlocal", "fde", &curwin->w_p_fde, 0) == FAIL
-      || put_setstring(fd, "setlocal", "fmr", &curwin->w_p_fmr, 0) == FAIL
-      || put_setstring(fd, "setlocal", "fdi", &curwin->w_p_fdi, 0) == FAIL
-      || put_setnum(fd, "setlocal", "fdl", &curwin->w_p_fdl) == FAIL
-      || put_setnum(fd, "setlocal", "fml", &curwin->w_p_fml) == FAIL
-      || put_setnum(fd, "setlocal", "fdn", &curwin->w_p_fdn) == FAIL
-      || put_setbool(fd, "setlocal", "fen", curwin->w_p_fen) == FAIL) {
+  if (put_set(fd, "setlocal", kOptFoldmethod, &curwin->w_p_fdm) == FAIL
+      || put_set(fd, "setlocal", kOptFoldexpr, &curwin->w_p_fde) == FAIL
+      || put_set(fd, "setlocal", kOptFoldmarker, &curwin->w_p_fmr) == FAIL
+      || put_set(fd, "setlocal", kOptFoldignore, &curwin->w_p_fdi) == FAIL
+      || put_set(fd, "setlocal", kOptFoldlevel, &curwin->w_p_fdl) == FAIL
+      || put_set(fd, "setlocal", kOptFoldminlines, &curwin->w_p_fml) == FAIL
+      || put_set(fd, "setlocal", kOptFoldnestmax, &curwin->w_p_fdn) == FAIL
+      || put_set(fd, "setlocal", kOptFoldenable, &curwin->w_p_fen) == FAIL) {
     return FAIL;
   }
 
   return OK;
 }
 
-static int put_setstring(FILE *fd, char *cmd, char *name, char **valuep, uint64_t flags)
+/// Print the ":set" command to set a single option to file.
+///
+/// @param  fd       File descriptor.
+/// @param  cmd      Command name.
+/// @param  opt_idx  Option index in options[] table.
+/// @param  varp     Pointer to option variable.
+///
+/// @return FAIL on error, OK otherwise.
+static int put_set(FILE *fd, char *cmd, OptIndex opt_idx, void *varp)
 {
-  if (fprintf(fd, "%s %s=", cmd, name) < 0) {
-    return FAIL;
+  OptVal value = optval_from_varp(opt_idx, varp);
+  vimoption_T *opt = &options[opt_idx];
+  char *name = opt->fullname;
+  uint64_t flags = opt->flags;
+
+  if ((opt->indir & PV_BOTH) && varp != opt->var
+      && optval_equal(value, get_option_unset_value(opt_idx))) {
+    // Processing unset local value of global-local option. Do nothing.
+    return OK;
+  }
+
+  switch (value.type) {
+  case kOptValTypeNil:
+    abort();
+  case kOptValTypeBoolean: {
+    assert(value.data.boolean != kNone);
+    bool value_bool = TRISTATE_TO_BOOL(value.data.boolean, false);
+
+    if (fprintf(fd, "%s %s%s", cmd, value_bool ? "" : "no", name) < 0) {
+      return FAIL;
+    }
+    break;
+  }
+  case kOptValTypeNumber: {
+    if (fprintf(fd, "%s %s=", cmd, name) < 0) {
+      return FAIL;
+    }
+
+    OptInt value_num = value.data.number;
+    OptInt wc;
+
```

Comment:
Why is there now a blank line between `OptInt wc` and `wc_use_keyname(varp, &wc)`?

Reply:
Was just a stylistic choice to make it easier to read for myself. Will remove it since it's problematic

Reply:
Then it's inconsistent with the boolean and string branches. Actually, I was suggesting moving the blank line before the `OptInt wc`.

Reply:
Should be good now